File: platforms/windows/KhandobaSecureDocs/Config/EnvironmentConfig.cs
//
//  EnvironmentConfig.cs
//  Khandoba Secure Docs - Windows
//
//  Environment-specific configuration
//

namespace KhandobaSecureDocs.Config
{
    public enum Environment
    {
        Development,
        Test,
        Production
    }

    public static class EnvironmentConfig
    {
#if DEBUG
        private static readonly Environment CurrentEnvironment = Environment.Development;
#elif TEST
        private static readonly Environment CurrentEnvironment = Environment.Test;
#else
        private static readonly Environment CurrentEnvironment = Environment.Production;
#endif

        public static Environment Current => CurrentEnvironment;

        public static bool IsDevelopment => CurrentEnvironment == Environment.Development;
        public static bool IsTest => CurrentEnvironment == Environment.Test;
        public static bool IsProduction => CurrentEnvironment == Environment.Production;

        public static string GetSupabaseUrl()
        {
            return CurrentEnvironment switch
            {
                Environment.Development => "https://uremtyiorzlapwthjsko.supabase.co", // Dev project
                Environment.Test => "https://uremtyiorzlapwthjsko.supabase.co", // Test project
                Environment.Production => "https://uremtyiorzlapwthjsko.supabase.co", // Prod project
                _ => AppConfig.SupabaseUrl
            };
        }

        public static string GetSupabaseAnonKey()
        {
            return CurrentEnvironment switch
            {
                Environment.Development => "YOUR_DEV_SUPABASE_ANON_KEY",
                Environment.Test => "YOUR_TEST_SUPABASE_ANON_KEY",
                Environment.Production => "YOUR_PROD_SUPABASE_ANON_KEY",
                _ => AppConfig.SupabaseAnonKey
            };
        }

        public static bool EnableLogging => CurrentEnvironment != Environment.Production;
        public static bool EnableAnalytics => CurrentEnvironment != Environment.Development;
        public static bool EnableCrashReporting => CurrentEnvironment != Environment.Development;
        public static bool EnablePushNotifications => true;
        public static bool RequireBiometricAuth => CurrentEnvironment != Environment.Development;

        public static int SessionTimeoutMinutes => CurrentEnvironment switch
        {
            Environment.Development => 60,
            Environment.Test => 30,
            Environment.Production => 30,
            _ => 30
        };
    }
}

File: platforms/apple/Khandoba Secure Docs/Config/EnvironmentConfig.swift
//
//  EnvironmentConfig.swift
//  Khandoba Secure Docs
//
//  Environment-specific configuration based on build configuration
//

import Foundation

enum Environment: String {
    case development
    case test
    case production
    
    static var current: Environment {
        #if DEVELOPMENT
        return .development
        #elseif TEST
        return .test
        #else
        return .production
        #endif
    }
    
    var name: String {
        switch self {
        case .development: return "Development"
        case .test: return "Test"
        case .production: return "Production"
        }
    }
    
    var bundleIdentifier: String {
        switch self {
        case .development: return "com.khandoba.securedocs.dev"
        case .test: return "com.khandoba.securedocs.test"
        case .production: return "com.khandoba.securedocs"
        }
    }
    
    // iOS-ONLY: Using CloudKit exclusively - no Supabase configuration needed
    
    // Feature Flags
    var enableLogging: Bool {
        switch self {
        case .development, .test: return true
        case .production: return false
        }
    }
    
    var enableAnalytics: Bool {
        switch self {
        case .development: return false
        case .test, .production: return true
        }
    }
    
    var enableCrashReporting: Bool {
        switch self {
        case .development: return false
        case .test, .production: return true
        }
    }
    
    var enablePushNotifications: Bool {
        return true // Enabled for all environments
    }
    
    // Security Settings
    var sessionTimeoutMinutes: Int {
        switch self {
        case .development: return 60 // Longer timeout for dev
        case .test: return 30
        case .production: return 30
        }
    }
    
    var requireBiometricAuth: Bool {
        switch self {
        case .development: return false // Disable for easier testing
        case .test, .production: return true
        }
    }
}

struct EnvironmentConfig {
    static let current = Environment.current
    
    static var isDevelopment: Bool {
        current == .development
    }
    
    static var isTest: Bool {
        current == .test
    }
    
    static var isProduction: Bool {
        current == .production
    }
}
